### PR TITLE
06_As a user, I need to modify my announcement

### PR DESCRIPTION
**Pull request by Jean-Beru**
Issue #2

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 1]
```diff
@@ -59,9 +61,12 @@ public function previewAction(Job $job)
      * @Route("/{country}/{contract}/{slug}/update", name="job_update")
      * @ParamConverter("job", options={"mapping"={"country"="country","contract"="contract","slug"="slug"}})
      * @Template
+     * @Security("is_granted('IS_AUTHENTICATED_FULLY')")
      */
     public function updateAction(Request $request, Job $job)
     {
+        $this->denyAccessUnlessGranted(JobVoter::EDIT, $job);
```
Comment (iamluc) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Why not put this condition in the `@Security` annotation ?

File: src/SensioLabs/JobBoardBundle/Entity/User.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Entity/User.php]: type `guid` ?

File: src/SensioLabs/JobBoardBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -21,5 +21,14 @@
         <service id="form.transformer.html_purifier" class="Exercise\HTMLPurifierBundle\Form\HTMLPurifierTransformer">
             <argument type="service" id="exercise_html_purifier.default" />
         </service>
+
+        <service id="sensiolabs_connect.repository.user" class="SensioLabs\JobBoardBundle\Entity\UserRepository" factory-service="doctrine" factory-method="getRepository">
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/config/services.xml]: `sensiolabs_connect` ? or job_board ?

File: src/SensioLabs/JobBoardBundle/Resources/config/services.xml [hunk 4]
```diff
@@ -21,5 +21,14 @@
...
+            <argument>SensioLabsJobBoardBundle:User</argument>
+        </service>
+
+        <service id="security.access.job_voter" class="SensioLabs\JobBoardBundle\Security\Authorization\Voter\JobVoter" public="false">
+            <argument type="service" id="service_container" strict="false" />
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/config/services.xml]: do you really need the whole container here ?

File: src/SensioLabs/JobBoardBundle/TestsFunctional/JobBoardWebTestCase.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+use Symfony\Component\BrowserKit\Cookie;
+use SensioLabs\Connect\Security\Authentication\Token\ConnectToken;
+
+class JobBoardWebTestCase extends WebTestCase
+{
+    protected $client = null;
+
+    public function setUp()
+    {
+        $this->client = static::createClient();
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/TestsFunctional/JobBoardWebTestCase.php]: you're not destroying your objects in the `tearDown` method

File: src/SensioLabs/JobBoardBundle/TestsFunctional/JobBoardWebTestCase.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            'SensioLabs\JobBoardBundle\DataFixtures\ORM\LoadJob',
+        ];
+        $this->fixtures = $this->loadFixtures($classes)->getReferenceRepository();
+    }
+
+    protected function tearDown()
+    {
+        $this->client = null;
+        $this->fixtures = null;
+        $this->loadFixtures([]);
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/TestsFunctional/JobBoardWebTestCase.php]: this is not needed.

**Comment by salahm** [comment]
:+1:

**Pull request merged**